datastore/gcloud/aio/datastore/filter.py

```diff
@@ -0,0 +1,115 @@
+from typing import Any
+from typing import Dict
+from typing import List
+
+from gcloud.aio.datastore.constants import CompositeFilterOperator
+from gcloud.aio.datastore.constants import PropertyFilterOperator
+from gcloud.aio.datastore.value import Value
+
+
+class BaseFilter:
+    json_key: str
+
+    # pylint: disable=unused-argument
+    def __new__(cls, *args, **kwargs):
```

Comment:
Looks like `__new__` doesn't take any additional args -- you can remove `args` and `kwargs` here (as well as the pylint disable).

Reply:
That's what I did at the beginning, and I got `Unexpected keyword argument 'operator' in constructor call (unexpected-keyword-arg)` on `cls(operator=operator, filters=filters)` in CompositeFilter. And a similar one in PropertyFilter. Pycharm also complains about `__init__`'s signature being incompatible with `__new__`'s

So this is kind of containing the problem... I'm starting to think that this is just not worth it and that if we don't want to do this, we should probably just remove this method. As much as I would like to learn exactly how init and new work in Python, this might not be the best use of my time :)

Reply:
Interesting.. yeah, I agree, this is not worth your time right now. Let's just nuke the `__new__` / `__init__` methods on the abstract base classes entirely and rely on the fact that they're not publicly exposed.